pkg/pillar/rootfs/init.sh

```diff
@@ -6,20 +6,19 @@
 echo 'nameserver 8.8.8.8' > /etc/resolv.conf
 
 # Need to disable H/W TCP offload since it seems to mess us up
-for i in `cd /sys/class/net ; echo eth*` ; do
-  ethtool -K $i gro off
-  ethtool -K $i sg off
+for i in $(cd /sys/class/net || return ; echo eth*) ; do
+  ethtool -K "$i" gro off
+  ethtool -K "$i" sg off
 done
 
 # Need this for logrotate
 /usr/sbin/crond -d 8
 
 # Finally, we need to start Xen
-XENCONSOLED_ARGS='--log=all --log-dir=/var/log/xen' /etc/init.d/xencommons start
+# In case it hangs and we have no hardware watchdog we run it in the background
+XENCONSOLED_ARGS='--log=all --log-dir=/var/log/xen' /etc/init.d/xencommons start &
+sleep 5 # Let it come up
 
-# This is an optional component - only run it if it is there
-if [ -f /opt/zededa/bin/device-steps.sh ]; then
-    /opt/zededa/bin/device-steps.sh >/var/log/device-steps.log 2>&1
-fi
+echo 'Starting device-steps'
+/opt/zededa/bin/device-steps.sh >/var/log/device-steps.log 2>&1
 
-tail -f /dev/null /var/log/*.log
```

Comment:
Any reason you had to remove the last line @eriknordmark ? It has a tiny advantage of capturing non device-steps.log in log files managed by linuxkit AND it is the line that keeps at least one process in the foreground. We had some issues with linuxkit init system when containers only have background processes.